src/tests/userDetails.test.js

```diff
@@ -0,0 +1,298 @@
+          postalCode: '0000',
+          accountName: 'Equity',
+          country: 'Rwanda',
+          streetAdress: '',
+        });
+      expect(res).to.have.status(406);
+      expect(res.body).to.have.property('error');
+      const resDel = await chai
+        .request(app)
+        .post(`/users/profile/${user.id}`)
+        .set({ Authorization: 'Bearer 321910398420cwdweded2qd2dw' })
+        .send({
+          gender: 'male',
+          currency: 'RWF',
+          lang: 'Kinya',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+1234567890',
+          postalCode: '0000',
+          accountName: 'BK',
+          country: 'Rwanda',
+          streetAdress: 'KN109',
+        });
+      chai.expect(resDel).to.have.status(400);
+    });
+  });
+  let newUser;
+  let newToken;
+  let newUserId;
+  describe('User Details check for different user', function () {
+    before(async function () {
+      await sequelize.getQueryInterface().bulkInsert(
+        'users',
+        [
+          {
+            id: uuidv4(),
+            username: 'testaxxtesting',
+            email: 'testaxx@testing.com',
+            password: await hashPassword('Qwert@12345'),
+            createdAt: new Date(),
+            updatedAt: new Date(),
+          },
+        ],
+        {}
+      );
+      newUser = await userServices.getUserByEmail('testaxx@testing.com');
+      newUserId = newUser.dataValues.id;
+      const res = await chai.request(app).post('/users/login').send({
+        email: newUser.dataValues.email,
+        password: 'Qwert@12345',
+      });
+      newToken = res.body.token;
+    });
+    it('should create user details if not exists', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${newUserId}`)
+        .set({ Authorization: `Bearer ${newToken}` })
+        .send({
+          gender: 'female',
+          currency: 'RWf',
+          lang: 'Italian',
+          dateOfBirth: '2000-01-01',
+          placeOfLiving: 'York',
+          tel: '+336638956',
+          postalCode: '0000',
+          accountNumber: '12334',
+          accountName: 'small tet',
+          country: 'Burundi',
+          streetAdress: 'KN rel',
+        });
+      expect(res).to.have.status(200);
+      expect(res.body).to.have.property('message', 'Successfully updated');
+    });
+    it('should Update user Details obeying the model', async function () {
+      const res = await chai
+        .request(app)
+        .post(`/users/profile/${newUserId}`)
+        .set({ Authorization: `Bearer ${newToken}` })
+        .send({
+          gender: 'male',
+          currency: 'USD',
+          lang: 'test lang',
+          dateOfBirth: '1990-01-01',
+          placeOfLiving: 'New York',
+          tel: '+870899889',
+          postalCode: '0000',
+          accountNumber: '12334',
+          accountName: 'kim',
+          country: 'tzd',
+          streetAdress: 'KN107',
```

Comment:
Sometimes you are sending the same keys and values, if you had created one object it would reduce the lines of code and improve readability.

Reply:
not the same